test/runtime/fqdn.go

```diff
@@ -778,14 +782,94 @@ INITSYSTEM=SYSTEMD`
 		})
 	})
 
+	It("DNS proxy policy works if Cilium stops", func() {
+		targetURL := "http://world1.cilium.test"
+		targetIP := worldIps[WorldHttpd1]
+		invalidURL := "http://world1.outside.test"
+		invalidIP := outsideIps[OutsideHttpd1]
+
+		policy := `
+[
+	{
+		"labels": [{
+			"key": "dns-proxy"
+		}],
+		"endpointSelector": {
+			"matchLabels": {
+				"container:id.app1": ""
+			}
+		},
+		"egress": [
+			{
+				"toPorts": [{
+					"ports":[{"port": "53", "protocol": "ANY"}],
+					"rules": {
+						"dns": [
+							{"matchPattern": "*.cilium.test"}
+						]
+					}
+				}]
+			},
+			{
+				"toFQDNs": [{
+					"matchName": "world1.cilium.test"
+				}]
+			}
+		]
+	}
+]`
+		_, err := vm.PolicyRenderAndImport(policy)
+		Expect(err).To(BeNil(), "Policy cannot be imported")
+
+		expectFQDNSareApplied()
+
+		By("Curl from %q to %q", helpers.App1, targetURL)
+		res := vm.ContainerExec(helpers.App1, helpers.CurlFail(targetURL))
+		res.ExpectSuccess("Cannot connect from app1")
+
+		By("Curl from %q to %q should fail", helpers.App1, invalidURL)
+		res = vm.ContainerExec(helpers.App1, helpers.CurlFail(invalidURL))
+		res.ExpectFail("Can connect from app1 when it should not work")
+
+		By("Stopping Cilium")
+
+		defer func() {
+			// Defer a Cilium restart to make sure that keep started when test finished.
+			_ = vm.ExecWithSudo("systemctl start cilium")
+			vm.WaitEndpointsReady()
+		}()
+
+		res = vm.ExecWithSudo("systemctl stop cilium")
+		res.ExpectSuccess("Failed trying to stop cilium via systemctl")
+
+		By("Testing connectivity to the IP %q without DNS request", targetIP)
+		res = vm.ContainerExec(helpers.App1, helpers.CurlFail("http://%s", targetIP))
+		res.ExpectSuccess("Cannot connect app1")
+
+		By("Curl from %q to %q with Cilium down", helpers.App1, targetURL)
+		res = vm.ContainerExec(helpers.App1, helpers.CurlFail(targetURL))
+		res.ExpectFail("This request should fail because no dns-proxy")
+
+		By("Testing invalid connectivity to the IP %q without DNS request", invalidIP)
```

Comment:
This fails because the DNS proxy is down, right? not because the IP isn't in the policy? The reason I ask is that we're using `OutsideHttpd1` but that's pointing to one of the world*.cilium.test httpd images, no? (or does it actually make different containers?)